lib/src/migrators/module/references.dart

```diff
@@ -0,0 +1,326 @@
+// Copyright 2019 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+// The sass package's API is not necessarily stable. It is being imported with
+// the Sass team's explicit knowledge and approval. See
+// https://github.com/sass/dart-sass/issues/236.
+import 'package:sass/src/ast/sass.dart';
+import 'package:sass/src/visitor/recursive_ast.dart';
+
+import 'package:collection/collection.dart';
+
+import 'bidirectional_map.dart';
+import 'scope.dart';
+import '../../utils.dart';
+
+/// A mapping between member declarations and references.
+///
+/// The first pass of the module migrator generates this object. It then
+/// uses the information stored here on the second pass to determine how a
+/// member is used.
+class References {
+  References._();
+
+  /// Map between variable references and their declarations.
+  ///
+  /// This map is frozen after it is built to prevent further modification.
+  final variables = BidirectionalMap<VariableExpression,
+      SassNode /*VariableDeclaration|Argument*/ >();
+
+  /// Map between variable reassignments and the original declaration they
+  /// override.
+  ///
+  /// If a variable is reassigned multiple times, all reassignments will map
+  /// to the original declaration, not the previous reassignment.
+  ///
+  /// This map is frozen after it is built to prevent further modification.
+  final variableReassignments = BidirectionalMap<VariableDeclaration,
+      SassNode /*VariableDeclaration|Argument*/ >();
+
+  /// Map between mixin references and their declarations.
+  ///
+  /// This map is frozen after it is built to prevent further modification.
+  final mixins = BidirectionalMap<IncludeRule, MixinRule>();
+
+  /// Map between normal function references and their declarations.
```

Comment:
This doesn't include references to built-in functions, correct? Either way, it would be good to document that.